pyzo/pyzokernel/magic.py

```diff
@@ -108,23 +110,43 @@ def _convert_command(self, line):
         # Get interpreter
         interpreter = sys._pyzoInterpreter
         
-        # Check if it is a variable
-        command = line.rstrip()
-        if ' ' not in command:
-            if command in interpreter.locals:
+        # Check that line is not some valid python input
+        try :
+            # gets a list of 5-tuples, of which [0] is the type of token and [1] is the token string
+            ltok = list(tokenize.tokenize(io.BytesIO(line.encode('utf-8')).readline))
+        except tokenize.TokenError :  # typically this means an unmatched parenthesis
```

Comment:
I appreciate the effort to use the tokenize function. Unfortunately, this will fail on 2.x, where there is no BytesIO (at least not on 2.6 and below), and `tokenize()` works differently too.

Reply:
Ah. According to the [Python2.6 doc](https://docs.python.org/2.6/library/tokenize.html), there is still some hope with StringIO instead (see example). But as I am completely incompetent in Python2, I just added a version check and fell back to the old behaviour for Python2 (maybe this can be refined to some Python2.x).

Isn't Python 2.6 outdated anyway ?

Reply:
>  I am completely incompetent in Python2

I can only applaud that :)

Nevertheless, one of the ideas of Pyzo is that you can run any Python interpreter on your system. Also Pypy and Jython. And also old versions: it goes as low as Python 2.4 :D